store/sqlstore/retention_policy_store.go

```diff
@@ -329,118 +335,178 @@ func (s *SqlRetentionPolicyStore) Patch(patch *model.RetentionPolicyWithTeamAndC
 	return &newPolicy, nil
 }
 
-func (s *SqlRetentionPolicyStore) buildGetPolicyQuery(id string) (query string, props map[string]interface{}) {
+func (s *SqlRetentionPolicyStore) buildGetPolicyQuery(id string) (string, []interface{}, error) {
 	return s.buildGetPoliciesQuery(id, 0, 1)
 }
 
 // buildGetPoliciesQuery builds a query to select information for the policy with the specified
 // ID, or, if `id` is the empty string, from all policies. The results returned will be sorted by
 // policy display name and ID.
-func (s *SqlRetentionPolicyStore) buildGetPoliciesQuery(id string, offset, limit int) (query string, props map[string]interface{}) {
-	props = map[string]interface{}{"Offset": offset, "Limit": limit}
-	whereIdEqualsPolicyId := ""
+func (s *SqlRetentionPolicyStore) buildGetPoliciesQuery(id string, offset, limit int) (string, []interface{}, error) {
+	rpcSubQuery := s.getQueryBuilder().
+		Select("RetentionPolicies.Id, COUNT(RetentionPoliciesChannels.ChannelId) AS Count").
+		From("RetentionPolicies").
+		LeftJoin("RetentionPoliciesChannels ON RetentionPolicies.Id = RetentionPoliciesChannels.PolicyId").
+		GroupBy("RetentionPolicies.Id").
+		OrderBy("RetentionPolicies.DisplayName, RetentionPolicies.Id").
+		Limit(uint64(limit)).
+		Offset(uint64(offset))
```

Comment:
Let's refactor this to a common part and apply them to `rpcSubQuery` and `rptSubQuery`.

Reply:
I didn't quite get this comment.. `rpcSubQuery` and `rptSubQuery` are similar but they're counting and joining on different columns from different tables `RetentionPoliciesChannels.ChannelId - RetentionPoliciesTeams.TeamId`.

Reply:
I was suggesting to use an `sq.Expr` to refactor this part outside and apply that on both the queries. But I see that it runs into the args problem again. Let's leave it for now.